src/persist-client/src/internal/state_diff.rs

```diff
@@ -66,6 +66,7 @@ pub struct StateDiff<T> {
     pub(crate) applier_version: semver::Version,
     pub(crate) seqno_from: SeqNo,
     pub(crate) seqno_to: SeqNo,
+    pub(crate) hostname: String,
```

Comment:
hmmm just realized that we probably want to store hostname as diffs, otherwise we lose the IMO nice property that we can turn persist's internal state into collections without something like upsert

it seems silly to plumb a whole new StateFieldDiff for this. maybe I can just lump it in with last_gc_req and make a struct of unrelated small things that don't change all that often? also possible I should figure out a way to make it less boilerplate to add a new StateFieldDiff type

Reply:
how painful is the boilerplate? it seems nice to treat this field as any other, even if we don't expect it to be called in the code anywhere

Reply:
probably more painful than it should be, but I think you're right. will give this a go

Reply:
actually wasn't too bad!